### PR TITLE
SDL Core implementation of feature - 0266 New vehicle data gearStatus

### LINKED ISSUE #3216: [SDL 0266] New vehicle data GearStatus
Proposal: [New vehicle data GearStatus](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0266-New-vehicle-data-GearStatus.md)

> This proposal is to add new vehicle data `GearStatus` to know drive mode and actual gear the transmission has selected.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/881

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the following revisions: use `userSelectedGear` in place of `selectedGear` and use the descriptions  for `GearStatus` included in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/881#issuecomment-573838656).
> The proposal .md file was updated to reflect these revisions on 1/21/2020.

### PR DESCRIPTION
**Pull request by mkorniichuk**
Fixes #3216 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
* Unit testing
* Manual testing
* ATF testing

### Summary
Current PR contains implementation of new vehicle data type gearStatus.

__Note:__ please, be noticed, the latest commit contains update related to rpc_spec submodule. So, it should be dropped before merge.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/appMain/sdl_preloaded_pt.json [hunk 1]
```diff
@@ -587,7 +587,8 @@
                             "odometer",
```
Comment (IGapchuk) [inline src/appMain/sdl_preloaded_pt.json:587]: @IGapchuk move to the DrivingCharacteristics-3 section

File: src/appMain/sdl_preloaded_pt.json [hunk 2]
```diff
@@ -3403,7 +3407,44 @@
                     "name": "prndl",
                     "key": "OEM_REF_PRNDL",
                     "type": "PRNDL",
-                    "mandatory": false
+                    "mandatory": false,
```
Comment (IGapchuk) [inline src/appMain/sdl_preloaded_pt.json:3410]: @IGapchuk move to a separate commit with comment that it is redundent

File: src/appMain/sdl_preloaded_pt.json (hunk 1 above)
Comment (IGapchuk) [reply src/appMain/sdl_preloaded_pt.json]: Done in the 8dbe04915c0c864bd1f429cd6901ea1c685f59e0

File: src/appMain/sdl_preloaded_pt.json (hunk 2 above)
Comment (IGapchuk) [reply src/appMain/sdl_preloaded_pt.json:3410]: Done in the https://github.com/smartdevicelink/sdl_core/pull/3401/commits/964eed04967e8a32b6a0109e93ce577f9666a420

**Comment by LitvinenkoIra** [comment]
@atiwari9 please review this PR

**Comment by atiwari9** [comment]
PR Approved.

Validated with atf scripts commit: [b4e023d](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2417/commits/b4e023d4204dc85a3b2889e6fd525512190a8837)

Logs:[Test_Report_Gear_Status_b4e023d_2.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4793880/Test_Report_Gear_Status_b4e023d_2.zip)

**Comment by LitvinenkoIra** [comment]
@theresalech PR is ready for Livio review. Thank you!

File: src/appMain/sdl_preloaded_pt.json [hunk 3]
```diff
@@ -3403,7 +3407,44 @@
...
+                    "since": "1.0",
+                    "until": "6.2"
+                },
+                {
```
Comment (ShobhitAd) [inline src/appMain/sdl_preloaded_pt.json:3414]: Could you please explain why two `prndl` vehicle data items are required.

File: src/appMain/sdl_preloaded_pt.json [hunk 4]
```diff
@@ -3403,7 +3407,44 @@
...
+                            "name": "transmissionType",
+                            "key": "OEM_REF_TR_TYPE",
+                            "type": "TransmissionType",
+                            "mandatory": false
+                        }
+                    ],
+                    "key": "OEM_REF_GEAR_STATUS",
+                    "type": "Struct",
+                    "mandatory": false,
+                    "since": "6.2"
```
Comment (ShobhitAd) [inline src/appMain/sdl_preloaded_pt.json:3447]:
```suggestion
                    "since": "7.0"
```

File: src/components/interfaces/HMI_API.xml [hunk 5]
```diff
@@ -6005,6 +6051,9 @@
     <param name="turnSignal" type="Boolean" mandatory="false">
         <description>See TurnSignal</description>
     </param>
+    <param name="gearStatus" type="Boolean" mandatory="false">
+      <description>See GearStatus</description>
+    </param>
     <param name="prndl" type="Boolean" mandatory="false">
       <description>See PRNDL</description>
```
Comment (ShobhitAd) [inline src/components/interfaces/HMI_API.xml:6058]: Should add a note for all `prndl` params stating that they are deprecated in the MOBILE API and that the param is covered by `gearStatus`

File: src/appMain/sdl_preloaded_pt.json (hunk 3 above)
Comment (LitvinenkoIra) [reply src/appMain/sdl_preloaded_pt.json]: @ShobhitAd I believe that we do not need the second one, removed in 89b14da7ace456037df1e73f9d262d18d850e573

**Comment by LitvinenkoIra** [comment]
@ShobhitAd Thank you for the review, all comments are processed, please check the latest updates.

File: src/appMain/sdl_preloaded_pt.json [hunk 6]
```diff
@@ -3403,7 +3407,36 @@
                     "name": "prndl",
                     "key": "OEM_REF_PRNDL",
                     "type": "PRNDL",
-                    "mandatory": false
+                    "mandatory": false,
+                    "since": "1.0",
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/appMain/sdl_preloaded_pt.json]: @ShobhitAd Thank you for the review, all comments are processed, please check the latest updates.

**Comment by ShobhitAd** [comment]
The RPC spec PR for this feature https://github.com/smartdevicelink/rpc_spec/pull/252 has been merged into develop. Please update the `.gitmodules` file to use the `smartdevicelink` remote and update the `tools/rpc_spec` submodule commit.

**Comment by AKalinich-Luxoft** [comment]
> The RPC spec PR for this feature [smartdevicelink/rpc_spec#252](https://github.com/smartdevicelink/rpc_spec/pull/252) has been merged into develop. Please update the `.gitmodules` file to use the `smartdevicelink` remote and update the `tools/rpc_spec` submodule commit.

@ShobhitAd done in https://github.com/smartdevicelink/sdl_core/pull/3401/commits/5495b462386a848542f152cd9a946e48b0ccbcdd
